lua/mini/starter.lua

```diff
@@ -524,29 +524,35 @@ MiniStarter.sections.recent_files = function(n, current_dir, show_path)
 
   return function()
     local section = string.format('Recent files%s', current_dir and ' (current directory)' or '')
-
-    -- Use only actual readable files
-    local files = vim.tbl_filter(function(f) return vim.fn.filereadable(f) == 1 end, vim.v.oldfiles or {})
-
-    if #files == 0 then
-      return { { name = 'There are no recent files (`v:oldfiles` is empty)', action = '', section = section } }
-    end
-
-    -- Possibly filter files from current directory
-    if current_dir then
-      local sep = vim.loop.os_uname().sysname == 'Windows_NT' and [[%\]] or '%/'
-      local cwd_pattern = '^' .. vim.pesc(vim.fn.getcwd()) .. sep
-      -- Use only files from current directory and its subdirectories
-      files = vim.tbl_filter(function(f) return f:find(cwd_pattern) ~= nil end, files)
+    local sep = vim.loop.os_uname().sysname == 'Windows_NT' and [[%\]] or '%/'
+    local cwd_pattern = '^' .. vim.pesc(vim.fn.getcwd()) .. sep
+
+    local n_added = 0
+    local files = {}
+    for _, f in ipairs(vim.v.oldfiles) do
+      if n_added == n then break end
+
+      -- Use only actual readable files
+      local add = vim.fn.filereadable(f) == 1
+      if current_dir then
+        -- Filter files from current directory and its subdirectories
+        add = add and f:find(cwd_pattern) ~= nil
+      end
+      if add then
+        n_added = n_added + 1
+        table.insert(files, f)
+      end
```

Comment:
This can be simplified

```suggestion
      if n_added == n then break end

      -- Use only actual readable files possibly respecting current directory
      local is_readable = vim.fn.filereadable(f) == 1
      local is_in_dir = not current_dir or f:find(cwd_pattern) ~= nil
      if is_readable and is_in_dir then
        n_added = n_added + 1
        table.insert(files, f)
      end

      if n_added >= n then break end
```

Reply:
I see an `if n_added` on both lines 533 and 534. 
Would `if n_added >= n` on only line 533 be sufficient?